### PR TITLE
Fix sporadic failures in MonoTests.System.Net.HttpRequestStreamTest on x64 Windows.

### PR DESCRIPTION
**Pull request by lateralusX**
MonoTests.System.Net.HttpRequestStreamTest suite has sporadic failures on Windows x64 with following exception:

System.Net.Sockets.SocketException : An operation was attempted on something that is not a socket.

This happens due to a data type and constant size mismatch. w32socket.h defines INVALID_SOCKET and SOCKET_ERROR for all platforms. On Windows x64, SOCKET is defined by winsock2.h as 64-bit type as well as INVALID_SOCKET constant, but above code will define INVALID_SOCKET as a 32-bit constant value. This will in turn cause code to sporadically fail to detect invalid sockets as returned by win32 API's. This will in turn cause the socket exception when code continues to use socket as argument to win32 socket API’s.

Current “invalid” x64 codegen for expression:

if (sock == INVALID_SOCKET)

mov         eax,0FFFFFFFFh
cmp         rsi,rax

and with fixed defines, correct 64-bit cmp instruction:

cmp         rsi,0FFFFFFFFFFFFFFFFh

Fix makes sure we get the defines of INVALID_SOCKET and SOCKET_ERROR from winsock2.h on Windows platforms to be consistent with define of SOCKET type.

While working on repro of the above error I also trapped another problem in the debugger that trigger the second commit in this PR:

Invalid handle exception during WSACleanup on shutdown.

Running the  MonoTests.System.Net.HttpRequestStreamTest under debugger reveals this problem. WSACleanup is freeing an invalid handle as part of network shutdown.

This happens since mono_w32socket_close calls incorrect win32 API when closing sockets. It currently calls CloseHandle, while it should call closesocket. This probably cause problems in winsock book keeping of active sockets, and when closing the network layer, winsock will try to close handles that it still thinks are open, but since the handles have already been closed by incorrect call to CloseHandle, it will close an “invalid handle” either causing the exception or even worse, close a different handle currently in use, causing undefined behavior during the rest of mono shutdown.

Fix is to switch to correct API when closing sockets on Windows, closesocket instead of CloseHandle.

Last commit is just adding some new files into correct filter folders in Visual Studio project, piggy backed into this PR.

### CODE REVIEW

**Comment by lateralusX** [comment]
Looks like a Jenkins environment error triggered for all win32 x86 builds at the moment:

hudson.plugins.git.GitException: Command "git clean -fdx" returned status code 1

**Comment by alexischr** [comment]
build

File: msvc/libmonoruntime.vcxproj.filters [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (baulig) [inline msvc/libmonoruntime.vcxproj.filters]: What are the changes in this file about?

File: msvc/libmonoutils.vcxproj.filters [hunk 2]
```diff
@@ -426,9 +417,9 @@
     <ClInclude Include="..\mono\utils\os-event.h">
       <Filter>Header Files</Filter>
     </ClInclude>
```
Comment (baulig) [inline msvc/libmonoutils.vcxproj.filters]: Unrelated changes.

File: msvc/libmonoruntime.vcxproj.filters (hunk 1 above)
Comment (lateralusX) [reply msvc/libmonoruntime.vcxproj.filters]: VS adds UTF8 BOM mark to files when saved from within environment. I can manually remove that.

File: msvc/libmonoutils.vcxproj.filters (hunk 2 above)
Comment (lateralusX) [reply msvc/libmonoutils.vcxproj.filters]: Also done by VS environment when project file was changed, I could manually revert that change outside the editor as well.

**Comment by lateralusX** [comment]
@baulig @luhenry Could we complete the reviews so this one can be merged?
